src/sonic-yang-models/yang-models/sonic-vlan.yang

```diff
@@ -2,189 +2,202 @@ module sonic-vlan {
+                        (contains(../ip-prefix, '.') and current()='IPv4')";
+                    type stypes:ip-family;
+                }
+            }
+            /* end of VLAN_INTERFACE_LIST */
+        }
+        /* end of VLAN_INTERFACE container */
 
-			list VLAN_LIST {
+        container VLAN {
 
-				key "name";
+            description "VLAN part of config_db.json";
+
+            list VLAN_LIST {
+
+                key "name";
         
-				leaf name {
-					type string {
-						pattern 'Vlan([0-9]{1,3}|[1-3][0-9]{3}|[4][0][0-8][0-9]|[4][0][9][0-4])';
-					}
-				}
-
-				leaf vlanid {
-					type uint16 {
-						range 1..4094;
-					}
-				}
-
-				leaf description {
-					type string {
-						length 1..255;
-					}
-				}
-
-				leaf-list dhcp_servers {
-					type inet:ip-address;
-				}
-
-				leaf mtu {
-					type uint16 {
-						range 1..9216;
-					}
-				}
-
-				leaf admin_status {
-					type stypes:admin_status;
-				}
-			}
-			/* end of VLAN_LIST */
-		}
-		/* end of container VLAN */
-
-		container VLAN_MEMBER {
-
-			description "VLAN_MEMBER part of config_db.json";
-
-			list VLAN_MEMBER_LIST {
-
-				key "name port";
-
-				leaf name {
-					type leafref {
-						path "/vlan:sonic-vlan/vlan:VLAN/vlan:VLAN_LIST/vlan:name";
-					}
-				}
-
-				leaf port {
-					/* key elements are mandatory by default */
-					type leafref {
-						path /port:sonic-port/port:PORT/port:PORT_LIST/port:name;
-					}
-				}
-
-				leaf tagging_mode {
-					mandatory true;
-					type stypes:vlan_tagging_mode;
-				}
-			}
+                leaf name {
+                    type string {
+                        pattern 'Vlan([0-9]{1,3}|[1-3][0-9]{3}|[4][0][0-8][0-9]|[4][0][9][0-4])';
+                    }
+                }
+
+                leaf vlanid {
+                    type uint16 {
+                        range 1..4094;
+                    }
+                }
+
+                leaf description {
+                    type string {
+                        length 1..255;
+                    }
+                }
+
+                leaf-list dhcp_servers {
+                    type inet:ip-address;
+                }
+
+                leaf mtu {
+                    type uint16 {
+                        range 1..9216;
+                    }
+                }
+
+                leaf admin_status {
+                    type stypes:admin_status;
+                }
+            }
+            /* end of VLAN_LIST */
+        }
+        /* end of container VLAN */
+
+        container VLAN_MEMBER {
+
+            description "VLAN_MEMBER part of config_db.json";
+
+            list VLAN_MEMBER_LIST {
+
+                key "name ifname";
+
+                leaf name {
+                    type leafref {
+                        path "/vlan:sonic-vlan/vlan:VLAN/vlan:VLAN_LIST/vlan:name";
+                    }
+                }
+
+                leaf ifname {
```

Comment:
please do not change the name here.

Reply:
In community call regarding naming, we have decided to name it as port for physical ethernet port and ifname for mix of ports cases. And just “name” for interfaces defined in sonic-port.yang, sonic-portchannel.yang, sonic-vlan.yang. Also since this is a key attribute, changing attribute-name will not cause any backward compatibility impact. Based on these, we have updated here

Reply:
ifname for mix of ports cases.?, ..., ifname is for interface name not for mix of ports.

We can stick to port:
as https://github.com/Azure/sonic-buildimage/blob/29763cd095e8bb63c0756a84b375ef5d445f4648/src/sonic-yang-models/yang-models/sonic-acl.yang#L258

Reply:
i do not understand this. vlan_member is only for one member port, why do we have a list of port now?

Reply:
@lguohan : Same VLAN can have multiple ports as members. Also same VLAN can have ethernet port & port-channel as members here. The comment mix-of-ports is mentioned in the perspective of different types of interfaces (Ethernet, Port-channel). Hope this clarifies the context of discussion.

@praveen-li : I remember, while we were discussing about consistent naming for interface names for this PR (https://github.com/Azure/sonic-buildimage/pull/7049) in call, we said port is generally in the context specific to  "Physical ethernet interfaces" and ifname (since it's interface-name) can be widely used to refer multiple different types of interfaces like Ethernet & Port-channel together + other types of interfaces like VLAN, loopback in other applications context like BGP where it will enable on all types of interfaces. In those understanding, here we have renamed it as ifname to include Etherent & Port-channel together.